### PR TITLE
c101007016

### PR DESCRIPTION
**Pull request by LogicalNonsense2051**
Added "World Legacy - "World Ark"".

### CODE REVIEW

**Comment by EerieCode** [comment]
I've taken a look at the script: it's mostly alright, but I'd like to point out a few mistakes:
- At line 39, there's no GetPreviousType() function on Ygopro, and the Get functions always return the value without letting you compare it. A more appropriate syntax would be `c:GetPreviousTypeOnField() & TYPE_LINK ~= 0`
- For the third effect, you forgot to rename tgfilter in the target and operation, it should be dkfilter
- At line 77, there's no SUMMON_SUMMON constant: the correct one for Normal Summons is SUMMON_TYPE_NORMAL
- Finally, the target for the first effect is incomplete. The first effect specifies that you can target a Link Monster among the destroyed monsters, while the current script targets any Link monster in the GY. You should limit the check in the filter to the contents of eg: the script would look something like this:
`function s.spfilter(c,e,tp)
    return c:IsType(TYPE_LINK) and c:IsCanBeSpecialSummoned(e,0,tp,false,false) and c:IsCanBeEffectTarget(e)
end
function s.sptg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
    if chkc then return eg:IsContains(chkc) and s.spfilter(chkc,e,tp) end
    if chk==0 then return Duel.GetLocationCount(tp,LOCATION_MZONE)>0
        and eg:IsExists(s.spfilter,1,nil,e,tp) end
    Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_SPSUMMON)
    local g=eg:FilterSelect(tp,s.spfilter,1,1,nil,e,tp)
    Duel.SetTargetCard(g)
    Duel.SetOperationInfo(0,CATEGORY_SPECIAL_SUMMON,g,1,0,0)
end`

Also, not exactly a mistake, but feel free to leave s,id in future scripts: the scripts works pretty much the same, but it's cleaner-looking and easier to update and use as reference. Still, it should be fine even with the beta ID.

**Comment by LogicalNonsense2051** [comment]
Oh, wasn't aware that there was no such function. Now I understand why you did so in the script of Sunburn the Solar Calamity" (having "c:GetPreviousAttributeOnField() & ATTRIBUTE_FIRE ~=0" in its cfilter).

Somehow, I missed out the tgfilter a couple of times, thank you for pointing out the error. For SUMMON_SUMMON, I realize my mistake now with naming the constant. At the time, I was thinking of OCG terms (where summon would mean "normal summon", rather than all the possible summons), while forgetting to type in _TYPE_ in between. 

The issue relating to the first effect was a major failure, I knew that it could only target link monsters that were just destroyed, rather than any in the GY, but admittedly, I didn't really know how to do so. As for the ID substitution, I wasn't sure if it was fine to keep it in, as most other scripts just uses the beta ID. I will try to keep in mind of how it is another option, in terms of ID, to use for future scripts.

I appreciate the time taken for you to look it over and would like to thank you once more.

File: SAST-JP/c101007016.lua [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+local s,id=GetID()
+
+function c101007016.initial_effect(c)
+	--Special summon link monster, optional trigger effect
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(id,0))
+	e1:SetCategory(CATEGORY_SPECIAL_SUMMON+CATEGORY_DAMAGE)
+	e1:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_TRIGGER_O)
+	e1:SetCode(EVENT_DESTROYED)
+	e2:SetProperty(EFFECT_FLAG_DELAY+EFFECT_FLAG_CARD_TARGET)
```
Comment (ahtelel) [inline SAST-JP/c101007016.lua]: Should this perhaps be e1

**Comment by ahtelel** [comment]
Well done

**Pull request merged**